### PR TITLE
[Feature] Support <folder> option on npm publish

### PR DESCRIPTION
**Pull request by nathf**
## Description

Previously listed as wont fix in #91, but the npm command allows publishing from a specific directory which allows easier organisation for 'flat' module structures.
So this PR exposes the folder option to align with [npm publish](https://docs.npmjs.com/cli/publish) options.

This change is non-breaking.

Unsure if this aligns with how the maintainers want to proceed with this tool, thought this would be a good starting point.

## Motivation and Context

Align with npm options, simplify package structures.

## How Has This Been Tested?

Tested locally through npm link in another lerna repo. Appropriate errors are thrown if `publishDir` doesnt contain a package.json, won't fail silently.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by gauntface** [comment]
I would love this feature.

@hzoo is there any chance you could take a look at this?

**Comment by MattCain** [comment]
This would be awesome, at the moment I can't publish the built versions of my packages without a redundant `dist` directory at the root of the distribution.

File: README.md [hunk 1]
```diff
@@ -594,7 +594,8 @@ Running `lerna` without arguments will show all commands/options.
       "ignore": "component-*"
     }
   },
-  "packages": ["packages/*"]
+  "packages": ["packages/*"],
+  "publishDir": "lib/"
```
Comment (gauntface) [inline README.md]: Would it be worth changing this to `dist/` or `build/` it feels like a more common use case for developers

**Comment by gauntface** [comment]
One wrinkle I just realised.

How does this work with bootstrap?

If you use `prepublish` to build your project before everything is npm linked by `lerna bootstrap` you'll get in a messy / hard to reason with state as the ``package.json will need to be in the root of each package to find the `prepublish` npm script that would then generate the build directory.

My understanding is that this is how you should be using lerna, which means this folder option may actually cause confusion.

**Comment by nathf** [comment]
@gauntface Good point. During bootstrap symlinking happens before prepublish. So the build directory wouldn't exist at symlink time. 

If the order were reversed you could npm link from the build directory as it would need to contain a `package.json` for publishing.  Alternatively the build directory could be created during `preinstall` phase but those both seem a bit flakey/cause more confusion.

Also taking into account the addition of yarn workspaces this feature may be a stretch. 

Thoughts?

**Comment by gauntface** [comment]
Yeah, the more I thought about it, the more I realised this wouldn't fit my needs.

**Comment by alan-agius4** [comment]
Can we have this back please?

**Pull request closed without merging**